packages/vscode-extension/src/debug/taskTerminal/baseTaskTerminal.ts

```diff
@@ -0,0 +1,47 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as vscode from "vscode";
+
+const ControlCodes = {
+  CtrlC: "\u0003",
+};
+
+export abstract class BaseTaskTerminal implements vscode.Pseudoterminal {
+  protected writeEmitter = new vscode.EventEmitter<string>();
+  onDidWrite: vscode.Event<string> = this.writeEmitter.event;
+  protected closeEmitter = new vscode.EventEmitter<number>();
+  onDidClose?: vscode.Event<number> = this.closeEmitter.event;
+
+  constructor(private taskDefinition: vscode.TaskDefinition) {}
```

Comment:
Why taking `taskDefinition` as constructor input?

Reply:
Consider take `args` as parameter instead of `taskDefinition` since other information is not needed.

Reply:
The task definition contains all the custom properties in a task. (Now it is `command` and `args`). In the base class, can we keep all the custom properties?

Reply:
Make sense.